src/package.py

```diff
@@ -11,13 +11,16 @@
 from . import util
 from .builder import BuildDirectory, BuildHandle, HostBuilder, TargetBuilder
 
+from src.distfile import Distfile
+from src.version import SimpleVersion
+from typing import AbstractSet, Any, Callable, Dict, Optional, Set, Union
 log = logging.getLogger(__name__)
 
 
 class HostPackage:
-    def __init__(self, install_directory, name, version, homepage,
-                 build_depends, lib_depends, distfiles, build_cmd,
-                 resource_directory):
+    def __init__(self, install_directory: str, name: str, version: SimpleVersion, homepage: str,
+                 build_depends: Set['HostPackage'], lib_depends: Set['HostPackage'], distfiles: Dict[str, Distfile], build_cmd: Callable,
```

Comment:
Out of curiosity, there is no way to specify the typing of the callable?

Reply:
yes, there is... `Callable[args, return type]` or something like that...